examples/.gitignore

```diff
@@ -0,0 +1,3 @@
+*.o
```

Comment:
If we gitignore the artefacts it's not possible to `go install` or `go run` the examples, which I think is kind of neat.

Reply:
> There is already an example of per CPU map usage here: https://pkg.go.dev/github.com/cilium/ebpf#example-Map-PerCPU How can we make that more discoverable? Maybe you saw it but it didn't help?

Yes. This example uses pure Go to manipulate the percpu map, but I think a C example is also necessary for users who want to put the main logic in C files.

Reply:
@lmb Hi, after thinking carefully, I still think we should ignore all these intermediate files and instruct users on how to compile and build step by step. After all, `go run -exec sudo` hides many details.

Reply:
@lmb hello 😃